gwpy/timeseries/timeseries.py

```diff
@@ -1920,32 +1931,30 @@ def q_transform(self, qrange=(4, 64), frange=(0, numpy.inf),
 
         # build regular Spectrogram from peak-Q data by interpolating each
         # (Q, frequency) `TimeSeries` to have the same time resolution
-        nx = int(abs(Segment(*outseg)) / tres)
+        nx = int(abs(outseg) / tres)
```

Comment:
@alurban , why is this change necessary?

Reply:
@duncanmmacleod, the `outseg` object is already an instance of `~gwpy.segments.Segment`, which is enforced by the docstring, so the additional call to `Segment()` seemed redundant.

Reply:
Technically, the docstring enforces nothing, but I get your point. However, having this here gives the user a little more freedom to just pass in a simple `tuple`, rather than a `Segment`. Can you replace it?

Reply:
Thanks @duncanmmacleod, this is replaced in the latest commit.